runtime/fiber.c

```diff
@@ -353,8 +353,9 @@ CAMLprim value caml_ensure_stack_capacity(value required_space)
   Used by the GC to find roots on the stacks of running or runnable fibers.
 */
 
-Caml_inline int is_block_and_not_code_frag(value v) {
-  return Is_block(v) && caml_find_code_fragment_by_pc((char *) v) == NULL;
+Caml_inline int is_block_and_not_code_frag(scanning_action f, value v) {
+  return Is_block(v) &&
+      (caml_is_oldify_one(f) || caml_find_code_fragment_by_pc((char *) v) == NULL);
```

Comment:
There are repeated calls to `caml_is_oldify_one` in `caml_scan_stack`. In the ideal case, since this information doesn't change in a given call to `caml_scan_stack`, exactly one call is sufficient. Does the compiler already optimise it this way?

Reply:
@kayceesrk sorry for the naive question, but (assuming I mostly don't read assembly code) what's the equivalent of `-dcmm` for your favorite C compiler that I can use to answer this question?

Reply:
(With the new implementation using flags I'm very confident that the check will be optimized away, because it's not a function call to a separate object anymore. But it would still be nice to have a look.)